### DISCUSSION TITLE
libmesh_opt.so: undefined reference to ‘TIMPI:: ...’

### DISCUSSION THREAD

**Discussion by lipxisi** [idaholab/moose#25462] [Q&A General]
Holle

When I bulid libmesh, three functions were not found
/usr/bin/ld: ./.libs/libmesh_opt.so: undefined reference to TIMPI::Communicator::Communicator(int const&)' /usr/bin/ld: ./.libs/libmesh_opt.so: undefined reference to TIMPI::TIMPIInit::TIMPIInit(int, char const* const*, bool, bool, int)'
/usr/bin/ld: ./.libs/libmesh_opt.so: undefined reference to `TIMPI::Request::Request(int const&)'
collect2: error: ld returned 1 exit status

When I use ldd-r to show the dependency of libmesh_opt, three are undefinded
undefined symbol: _ZN5TIMPI12CommunicatorC2ERKi    (.libs/libmesh_opt.so)
undefined symbol: _ZN5TIMPI7RequestC1ERKi    (.libs/libmesh_opt.so)
undefined symbol: _ZN5TIMPI9TIMPIInitC1EiPKPKcbbi    (.libs/libmesh_opt.so)

Which library might lead to the problem,and how do I find out what the problem is.

Thank you very much for your suggestions and answers.

**Comment by milljm** [comment]
It would seem to me that perhaps some of libMesh's contribs were not updated (a submodule update within libmesh).
